ui/tests/e2e/login.test.ts

```diff
@@ -0,0 +1,76 @@
+import puppeteer from 'puppeteer'
+
+// const LOGIN_URL = 'http://localhost:3000/#/signin'
+const LOGIN_URL = process.env.CI
+  ? 'http://127.0.0.1:12333/dashboard/#/signin'
+  : 'http://localhost:3000/#/signin'
+
+describe('Login', () => {
+  let browser
+  beforeAll(async () => {
+    // browser = await puppeteer.launch({
+    //   headless: false,
+    //   slowMo: 100,
+    // })
+    browser = await puppeteer.launch()
+  })
+
+  afterAll(() => {
+    browser.close()
+  })
+
+  it(
+    'should login success by correct password',
+    async () => {
+      const page = await browser.newPage()
+
+      await page.goto(LOGIN_URL)
+
+      const title = await page.title()
+      expect(title).toBe('TiDB Dashboard')
+
+      await page.waitForSelector('button[type=submit]')
+      await page.click('button[type=submit]')
+
+      await page.waitForSelector('.ant-message-success')
+      const success = await page.$eval('.ant-message-success', (el) =>
+        el ? true : false
+      )
+      expect(success).toBe(true)
+      const content = await page.$eval(
+        '.ant-message-success span',
+        (el) => el.textContent
+      )
+      expect(content).toEqual('Sign in successfully')
+
+      // await page.screenshot({ path: 'screen-1.png' })
+    },
+    10 * 1000
+  )
+
+  it(
+    'should login fail by incorrect password',
+    async () => {
+      const page = await browser.newPage()
+      await page.goto(LOGIN_URL)
+
+      await page.waitForSelector('input#tidb_signin_password')
+      await page.type('input#tidb_signin_password', 'any')
+      await page.click('button[type=submit]')
+
+      await page.waitForSelector('.ant-form-explain')
+      const fail = await page.$eval('.ant-form-explain', (el) =>
+        el ? true : false
+      )
```

Comment:
relying on ant design's className is very risky, can we have better ways? We can enlarge the select target (i.e. selecting a parent element that have a fixed ID specified by us).

Reply:
tried but didn't find a very good way, I just replace it by judging 'div.has-error' element.